processor/runtime/src/main/java/com/redhat/service/smartevents/processor/actions/webhook/WebhookActionInvokerBuilder.java

```diff
@@ -1,31 +1,70 @@
 package com.redhat.service.smartevents.processor.actions.webhook;
 
-import java.util.Optional;
+import java.util.Objects;
 
 import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
 
-import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.auth.OidcClient;
+import com.redhat.service.smartevents.infra.auth.OidcClientConstants;
+import com.redhat.service.smartevents.infra.exceptions.definitions.platform.TechnicalBearerTokenNotConfiguredException;
 import com.redhat.service.smartevents.infra.models.dto.ProcessorDTO;
 import com.redhat.service.smartevents.infra.models.gateways.Action;
-import com.redhat.service.smartevents.processor.actions.AbstractWebClientInvokerBuilder;
 import com.redhat.service.smartevents.processor.actions.ActionInvoker;
+import com.redhat.service.smartevents.processor.actions.ActionInvokerBuilder;
+
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
 
 @ApplicationScoped
-public class WebhookActionInvokerBuilder extends AbstractWebClientInvokerBuilder implements WebhookAction {
+public class WebhookActionInvokerBuilder implements WebhookAction, ActionInvokerBuilder {
+
+    @Inject
+    Instance<OidcClient> oidcClients;
+
+    @Inject
+    Vertx vertx;
 
     @Override
     public ActionInvoker build(ProcessorDTO processor, Action action) {
-        String endpoint = Optional.ofNullable(action.getParameters().get(ENDPOINT_PARAM))
-                .orElseThrow(() -> buildNoEndpointException(processor));
-        return requiresTechnicalBearerToken(action)
-                ? new WebhookActionInvoker(endpoint, webClient, getOidcClient())
-                : new WebhookActionInvoker(endpoint, webClient);
+        String endpoint = action.getParameters().get(ENDPOINT_PARAM);
+
+        WebClientOptions options = isSslVerificationDisabled(action)
```

Comment:
@kostola I know you need this for the demo with the Ansible Tower instance using self-signed certs, so let's go with it for now.

In general, we should check that this is supported by InfoSec/ProdSec and also ensure that our Threat Modelling doesn't need to be updated to cover this. Turning off SSL verification always feels like a bad idea to me. It's _probably OK_ whilst the WebHook is fire and forget, but we should double check given that this is going to be running on our hosted infrastructure.

Can you please take a task to follow up with InfoSec/ProdSec on this change.

cheers,

Rob

Reply:
Follow-up task created: https://issues.redhat.com/browse/MGDOBR-718